### PR TITLE
Parth Shah - Task manager 2

### PR DESCRIPTION
**Pull request by parthshahva**
Completed all Employee requirements. I did quite a bit of refactoring to make client.rb have all the output code and no logic. Feeling accomplished.

### CODE REVIEW

**Comment by gilbert** [comment]
Your peer review partner is Brian Provost. He also completed phase two: https://github.com/makersquare/ruby-course/pull/106

File: client.rb [hunk 1]
```diff
@@ -0,0 +1,165 @@
+require_relative 'lib/task-manager.rb'
+
+class TM::ProjectManager
+  def self.start
+    tracker = TM::ProjectTracker.new
+    @@exit = false
+    while @@exit == false
+    puts "What would you like to do?"
+    puts "-- DOING THINGS"
```
Comment (provostprovost) [inline client.rb:9]: POST, POST, POST, PUT, PUT, PUT

oh this isn't a checkpoint, nm

File: client.rb [hunk 2]
```diff
@@ -0,0 +1,165 @@
...
+    puts "-- Type 'create NAME' - Create a new project with name=NAME"
+    puts "-- Type 'add PID DESC Priority' - Add a new task to project with id=PID"
```
Comment (provostprovost) [inline client.rb:11]: It probably makes more sense to structure this "add PID Priority DESC" to allow for multi-word descriptions, where desc = choice[3..-1]

File: client.rb [hunk 3]
```diff
@@ -0,0 +1,165 @@
...
+        when 'completed'
+            tasks = tracker.completed_employee_tasks(choice[1])
+              puts "Completed Tasks of employee with id #{choice[1]}"
+              puts "Task id: Description, Priority"
+              tasks.each do |x|
+                puts "#{x.id}: #{x.description}, #{x.priority_number}"
+              end
+        #GENERAL#
+        when 'help'
+          puts "The system is pretty straightforward. Figure it out!"
```
Comment (provostprovost) [inline client.rb:155]: Very helpful, thanks

File: lib/task-manager/employee.rb [hunk 4]
```diff
@@ -0,0 +1,16 @@
...
+  @@id_counter = 1
+  def initialize(name)
+    @name = name
+    @id = @@id_counter
+    @@id_counter += 1
+    @projects = []
+    @tasks = []
+  end
+
+  def self.id_counter=(value)
```
Comment (provostprovost) [inline lib/task-manager/employee.rb:13]: Does this do anything? It doesn't seem to change the default behavior of @@id_counter=.

File: lib/task-manager/task.rb [hunk 5]
```diff
@@ -0,0 +1,25 @@
...
+  @@id_counter = 1
+  def initialize(project_id, description, priority_number)
+    @project_id = project_id
+    @description = description
+    @priority_number = priority_number
+    @id = @@id_counter
+    @@id_counter += 1
+    @status = "incomplete"
+    @creation_date = Time.now
+    @employee = nil
```
Comment (provostprovost) [inline lib/task-manager/task.rb:15]: Nil is a terrible employee

Nilton: http://dashburst.com/wp-content/uploads/2013/08/milton-office-space.jpg

**Comment by provostprovost** [comment]
Looks good, except that we have to change everything on Monday!

**Pull request closed without merging**